hw8/todolist/src/components/WeatherWidget/WeatherWidget.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from "react";
+import "./WeatherWidget.css";
+
+const defaultLatitude = 41.6938,
+  defaultLongitude = 44.8015;
+
+const prepareData = (): Promise<any> => {
+  return getUserLocation()
+    .then((location: { latitude: number; longitude: number }) => {
+      return getWeatherData(location.latitude, location.longitude);
+    })
+    .catch(() => {
+      return getWeatherData(defaultLatitude, defaultLongitude);
+    });
+};
+
+const WeatherWidget = () => {
+  const [data, setData] = useState<any>(null);
+
+  useEffect(() => {
+    prepareData().then((result) => setData(result));
+  }, []);
+
+  if (!data) {
+    return <div></div>;
+  }
+  return (
+    <div className="row header__weather">
```

Comment:
You can add styles for the current component in the css file just for tags - it will be used only for this component, so you don't need to add long classnames